plugin/trino-hive/src/main/java/io/trino/plugin/hive/cache/HiveCacheKeyProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public final class HiveCacheKeyProvider
+        implements CacheKeyProvider
+{
+    @Override
+    public Optional<String> getCacheKey(TrinoInputFile inputFile)
+            throws IOException
+    {
+        // We use file length in the cache key instead of lastModified to avoid making extra FS calls to fetch last modified time of file.
+        // We make a simplifying assumption that files are not rewritten with exactly same path and size but different content.
+        return Optional.of(inputFile.location().path() + inputFile.length());
```

Comment:
Can we add some  sort of an splitter between the path and input file length - as it might leads to invalid cache lookups for file like `hdfs:///.../1` of length 111 and `hdfs:///.../11` of length 1 -

Reply:
added a `#` in between